integration-tests/src/tests/client/features/multinode_stateless_validators.rs

```diff
@@ -556,54 +369,63 @@ fn test_stateless_validators_with_multi_test_loop() {
         );
         *balances.get_mut(&accounts[i]).unwrap() -= amount;
         *balances.get_mut(&accounts[(i + 1) % NUM_ACCOUNTS]).unwrap() += amount;
-        drop(
-            test.sender()
-                .for_index(i % NUM_VALIDATORS)
-                .with_additional_delay(Duration::milliseconds(300 * i as i64))
-                .into_wrapped_multi_sender::<ClientSenderForNetworkMessage, ClientSenderForNetwork>(
-                )
-                .send_async(ProcessTxRequest {
-                    transaction: tx,
-                    is_forwarded: false,
-                    check_only: false,
-                }),
-        );
+        let _ = node_datas[i % NUM_VALIDATORS]
+            .client_sender
+            .clone()
+            .with_delay(Duration::milliseconds(300 * i as i64))
+            .send_async(ProcessTxRequest {
+                transaction: tx,
+                is_forwarded: false,
+                check_only: false,
+            });
     }
 
     // Run the chain some time to allow transactions be processed.
-    test.run_for(Duration::seconds(20));
+    test_loop.run_for(Duration::seconds(20));
 
     // Capture the id of the epoch we will check for the correct validator information in assert_validator_info.
-    let prev_epoch_id = test.data[0].client.client.chain.head().unwrap().epoch_id;
+    let prev_epoch_id = test_loop.data.get(&client_handle).client.chain.head().unwrap().epoch_id;
     assert_ne!(prev_epoch_id, initial_epoch_id);
 
     // Run the chain until it transitions to a different epoch then prev_epoch_id.
-    test.run_until(
-        |data| data[0].client.client.chain.head().unwrap().epoch_id != prev_epoch_id,
+    test_loop.run_until(
+        |test_loop_data| {
+            test_loop_data.get(&client_handle).client.chain.head().unwrap().epoch_id
+                != prev_epoch_id
+        },
         Duration::seconds(EPOCH_LENGTH as i64),
     );
 
     // Check that the balances are correct.
+    let clients = node_datas
```

Comment:
Note: We are using the change from AsRef<Client> to Borrow<Client> semantics here.